binding/form_mapping.go

```diff
@@ -163,8 +163,10 @@ func setByForm(value reflect.Value, field reflect.StructField, form map[string][
 			val = opt.defaultValue
 		}
 
-		if len(vs) > 0 {
+		if len(vs) > 0 && vs[0] != "" {
 			val = vs[0]
+		} else {
+			val = opt.defaultValue
 		}
```

Comment:
```go
val = opt.defaultValue
if len(vs) > 0 && vs[0] != "" {
	val = vs[0]
}
```

Reply:
```
		var val string
		if !ok {
			val = opt.defaultValue
		}
```
it's invalid code? it should update the follow code:
```
val := opt.defaultValue
```

Reply:
fixed discussion